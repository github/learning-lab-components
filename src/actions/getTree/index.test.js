const getTree = require('.')
const mockContext = require('../../tests/mockContext')

describe('getTree', () => {
  let context

  beforeEach(() => {
    context = mockContext({},
      {
        gitdata: {
          getRef: jest.fn(() => Promise.resolve({ data: { object: { sha: 1 } } })),
          getTree: jest.fn()
        }
      }
    )
  })

  it('calls github.gitdata.getTree', async () => {
    await getTree(context, {})
    expect(context.github.gitdata.getRef).toHaveBeenCalled()
    expect(context.github.gitdata.getTree).toHaveBeenCalled()
    expect(context.github.gitdata.getTree.mock.calls).toMatchSnapshot()
  })

  it('calls github.gitdata.getTree with recursive', async () => {
    await getTree(context, { recursive: true })
    expect(context.github.gitdata.getTree).toHaveBeenCalled()
    expect(context.github.gitdata.getTree.mock.calls).toMatchSnapshot()
  })

  it('calls github.gitdata.getTree with a custom sha', async () => {
    await getTree(context, { sha: 123 })
    expect(context.github.gitdata.getRef).not.toHaveBeenCalled()
    expect(context.github.gitdata.getTree).toHaveBeenCalled()
    expect(context.github.gitdata.getTree.mock.calls).toMatchSnapshot()
  })
})
