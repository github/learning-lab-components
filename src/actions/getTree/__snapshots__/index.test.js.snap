// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTree calls github.gitdata.getTree 1`] = `
Array [
  Array [
    Object {
      "owner": "JasonEtco",
      "repo": "example",
      "tree_sha": 1,
    },
  ],
]
`;

exports[`getTree calls github.gitdata.getTree with a custom sha 1`] = `
Array [
  Array [
    Object {
      "owner": "JasonEtco",
      "repo": "example",
      "tree_sha": 123,
    },
  ],
]
`;

exports[`getTree calls github.gitdata.getTree with recursive 1`] = `
Array [
  Array [
    Object {
      "owner": "JasonEtco",
      "recursive": 1,
      "repo": "example",
      "tree_sha": 1,
    },
  ],
]
`;
